nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -243,9 +243,13 @@ sub fingerprintUnit {
                             foreach my $socket (@sockets) {
                                 if (defined $activePrev->{$socket}) {
                                     $unitsToStop{$socket} = 1;
-                                    $unitsToStart{$socket} = 1;
-                                    recordUnit($startListFile, $socket);
-                                    $socketActivated = 1;
+                                    # Only restart sockets that actually
+                                    # exist in new configuration:
+                                    if (-e "$out/etc/systemd/system/$socket") {
+                                        $unitsToStart{$socket} = 1;
+                                        recordUnit($startListFile, $socket);
+                                        $socketActivated = 1;
+                                    }
```

Comment:
I am getting this error

```
building the system configuration...
trace: Warning: `stdenv.lib` is deprecated and will be removed in the next release. Please use `lib` instead. For more information see https://github.com/NixOS/nixpkgs/issues/108938
Global symbol "%unitsToStartFiltered" requires explicit package name (did you forget to declare "my %unitsToStartFiltered"?) at /nix/store/wp90mv7gm6fbdghnsmlyni3wnxzz7pnz-nixos-system-eve-21.05.20210312.fdc5e28/bin/switch-to-configuration line 462.
Execution of /nix/store/wp90mv7gm6fbdghnsmlyni3wnxzz7pnz-nixos-system-eve-21.05.20210312.fdc5e28/bin/switch-to-configuration aborted due to compilation errors.
warning: error(s) occurred while switching to the new configuration
building the system configuration...
Global symbol "%unitsToStartFiltered" requires explicit package name (did you forget to declare "my %unitsToStartFiltered"?) at /nix/store/wp90mv7gm6fbdghnsmlyni3wnxzz7pnz-nixos-system-eve-21.05.20210312.fdc5e28/bin/switch-to-configuration line 462.
Execution of /nix/store/wp90mv7gm6fbdghnsmlyni3wnxzz7pnz-nixos-system-eve-21.05.20210312.fdc5e28/bin/switch-to-configuration aborted due to compilation errors.
warning: error(s) occurred while switching to the new configuration
```

I am using nix flakes as well in case this makes a difference.

Reply:
I don't see how this error relate to the changes in this PR. It does not happen on master?
what is on line 462?

Reply:
Oh. Ignore this message. Might have been my own patch.